services/src/pro/datasets/in_memory.rs

```diff
@@ -446,7 +557,10 @@ impl
     ) -> Result<Box<dyn MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>>>
     {
         ensure!(
-            self.dataset_permissions
+            self.backend
+                .read()
+                .await
+                .dataset_permissions
```

Comment:
and here

Reply:
fccf47a2459c22ceb51d2ac1d1b4e0237d13cfd1